fission-cli/library/Fission/CLI/Handler/App/Publish.hs

```diff
@@ -104,17 +108,24 @@ publish watchFlag runner appURL appPath _updateDNS updateData = do -- TODO updat
               raise err
 
             Right _ ->
-              case watchFlag of
-                WatchFlag False ->
+              case (openFlag, watchFlag) of
+                (OpenFlag True, WatchFlag True) ->
                   success appURL
 
-                WatchFlag True ->
+                (OpenFlag True, _) -> do
+                  _ <- liftIO . openBrowser $ "https://ipfs.runfission.com/ipns/" <> (Text.unpack $ textDisplay appURL)
```

Comment:
This PR is listed as WIP, so totally get that this may be an initial sketch 👍 Some comments in the meantime:

Things stored on staging will _very likely_ be unavailable on production, so we should account for the environment. IIRC the `appURL` contains the full domain (e.g. staging vs prod), but if we're going to do the direct CID, there's a `class Fission.CLI.Remote` tells you which server environment you're aimed at. The type you'll get back (and some helpers) are available via [`Fission.Web.API.Remote`](https://github.com/fission-suite/fission/blob/main/fission-web-api/library/Fission/Web/API/Remote.hs)

Have you experimented with this at the `appURL` instead? It's possible that it's not available on DNS yet, but it's generally pretty quick. I would say this is preferable, but because of DNS, well YMMV. We could always check DNS first.

Reply:
So I factored this out into a couple IO actions above the main case statement and sequenced the actions to handle the `--open --watch` case...

>Things stored on staging will very likely be unavailable on production, so we should account for the environment. IIRC the appURL contains the full domain (e.g. staging vs prod), but if we're going to do the direct CID, there's a class Fission.CLI.Remote tells you which server environment you're aimed at. The type you'll get back (and some helpers) are available via Fission.Web.API.Remote

> Have you experimented with this at the appURL instead? It's possible that it's not available on DNS yet, but it's generally pretty quick. I would say this is preferable, but because of DNS, well YMMV. We could always check DNS first.

I was able to get the Environment stuff worked up but it wasn't quite as ergonomic as I'd have hoped. At the same time, I'm not really sure what I'd do instead either.

But Url doesn't have a `Scheme` and `BaseUrl`'s `showBaseUrl` function from Servant doesn't allow me to configure a subdomain so I just hacked together something that worked for now but it's not my favorite.

Reply:
You can get a subdomain in there:

```haskell
> BaseUrl Https "sub.domain.tld" 123 "a/b/c"
BaseUrl {baseUrlScheme = Https, baseUrlHost = "sub.domain.tld", baseUrlPort = 123, baseUrlPath = "a/b/c"}

> showBaseUrl $ BaseUrl Https "sub.domain.tld" 123 "a/b/c"
"https://sub.domain.tld:123/a/b/c"
```

Reply:
>  but it wasn't quite as ergonomic as I'd have hoped.

Tell me more about this?

Reply:
> get the Environment stuff

Sorry meaning opening to a browser that's pointed at a staging / prod environment url based on `getRemoteBaseUrl` from `MonadRemote` because of the subdomain piece & because it took me a while to find the `showBaseUrl` helper in Servant.Client.Core. I half expected the `Show` instance to work but when it didn't I had to dig a little deeper which was sort of annoying but then again, I can't really expect the `Show` instance to do anything but "pretty print" the data structure so not sure why I thought I'd get a sensible url string out of it.